client/my-sites/activity/filterbar/action-type-selector.jsx

```diff
@@ -24,15 +24,26 @@ export class ActionTypeSelector extends Component {
 		this.activityTypeButton = React.createRef();
 	}
 
-	renderCheckbox = checkbox => {
+	isAllCheckboxSelected = checkboxes => {
+		if ( ! this.props.selected ) {
+			return true;
+		}
+		if ( checkboxes.length === this.props.selected.length ) {
+			return true;
+		}
+
+		return false;
+	};
+
+	renderCheckbox = ( allCheckboxes, checkbox ) => {
 		const { onSelectClick, selected } = this.props;
 		return (
 			<FormLabel key={ checkbox.key }>
 				<FormCheckbox
 					id={ checkbox.key }
-					checked={ !! this.isSelected( selected, checkbox.key ) }
+					checked={ !! this.isSelected( selected, checkbox.key ) || ! selected }
 					name={ checkbox.key }
-					onChange={ onSelectClick.bind( this, checkbox ) }
+					onChange={ onSelectClick.bind( this, checkbox, allCheckboxes ) }
```

Comment:
I remember awhile ago, we decided to move away from using `bind` and I cannot remember why.

We can explore this in an other PR, but we likely do not need `bind` - we can get `checkbox` from event props, and we can make `allCheckboxes` part of the component state so it's readily available in all methods.

Reply:
I am not sure how to make this work without bind.